### DISCUSSION TITLE
feature request: support Response Mock plugin

### DISCUSSION THREAD

**Discussion by drery** [apache/apisix#5896] [Ideas]
插件功能：解析json schema格式，随机生成json格式的响应返回，不转发到后端

**Comment by juzhiyuan** [comment]
AFAIK, we could use `response-rewrite` plugin to customize response data, see https://apisix.apache.org/docs/apisix/plugins/response-rewrite

But it doesn't support generating random data according to the schema.  cc @tzssangglass to have a confirmation.

**Reply by drery** [reply]
期望的是绑定了mock插件后，就不再转发到upstream，response-rewrite插件似乎无法实现这一点。
这里是否可以参考request-validate插件解析body schema，开发一个Mock插件呢~

**Reply by drery** [reply]
btw, If I develop a mock plugin,I want to know if I can push the mock plugin to apisix

**Reply by tzssangglass** [reply]
we have `fault-injection` plugin: https://github.com/apache/apisix/blob/master/docs/en/latest/plugins/fault-injection.md, you can enhance in this plugin.

**Reply by tzssangglass** [reply]
or just use the serveless plugin,this function does not require a plugin to do.

**Reply by drery** [reply]
this idea is reference to kong mocking plugin: https://docs.konghq.com/hub/kong-inc/mocking

**Reply by juzhiyuan** [reply]
> btw, If I develop a mock plugin,I want to know if I can push the mock plugin to apisix

Of course! Welcome any kinds of contribution to Apache APISIX :)

**Reply by juzhiyuan** [reply]
> this idea is reference to kong mocking plugin: https://docs.konghq.com/hub/kong-inc/mocking

And cc @shuaijinchao

**Reply by shuaijinchao** [reply]
thanks @juzhiyuan , @Drery  has submitted a PR #5940 , and we will discuss this feature in the PR.

**Reply by juzhiyuan** [reply]
ok, also cc @guoqqqi @Baoyuantop to have a review #5940
